pkg/provider/azure_controller_common.go

```diff
@@ -620,6 +663,11 @@ func (c *controllerCommon) checkDiskExists(ctx context.Context, diskURI string)
 	return true, nil
 }
 
+func (c *controllerCommon) vmUpdateRequired(future *azure.Future, err error) bool {
+	errCode := getErrorCode(err)
```

Comment:
why not do string compare directly with `Code="OperationPreempted"`, that would be quite straightforward than current logic.

Reply:
I may get your point, it requires status code = 2xx, but if `Code="OperationPreempted"`, I think status code is always 2xx, and do vm update once does not harm, so only string compare is more straightforward IMO, but that way, you don't need following code:
```
func getErrorCode(err error) *string {
	if err == nil {
		return nil
	}
	matches := errorCodeRE.FindStringSubmatch(err.Error())
	if matches == nil {
		return nil
	}
	return &matches[1]
}

type FutureParser interface {
	ConfigAccepted(future *azure.Future) bool
}

type controllerFutureParser struct{}

func (c *controllerFutureParser) ConfigAccepted(future *azure.Future) bool {
	// if status code indicates success, the storage profile change was committed
	return future != nil && future.Response() != nil && future.Response().StatusCode/100 == 2
}
```

Reply:
We have observed other errors with status code of `2xx`, where we were instructed that VM Update needs to be retried until success. However, there were certain errors with code = `2xx`, where we couldn't recover by simply retrying.  The error code and future parsing was for future code extensibility so that we can address issues with other errors with `2xx`, where retry can solve this issue.